### PR TITLE
chore: Linter rule for properly formatted api errors

### PR DESCRIPTION
**Pull request by Emyrk**
All `Message` fields must be proper sentences.

### CODE REVIEW

File: scripts/rules.go [hunk 1]
```diff
@@ -115,3 +115,40 @@ func InTx(m dsl.Matcher) {
 	`).Where(m["x"].Text != m["y"].Text).
 		At(m["f"]).Report("Pass the tx database into the '$f' function inside the closure. Use '$y' over $x'")
 }
+
+// HttpAPIErrorMessage intends to enforce constructing proper sentences as
+// error messages for the api. A proper sentence includes proper capitalization
+// and ends with punctuation.
+// There are ways around the linter, but this should work in the common cases.
+// Example:
+//	Message:
```
Comment (kylecarbs) [inline scripts/rules.go:124]: We could probably remove this for now!

**Review by kylecarbs — APPROVED** [review]
This is awesome! One minor nit that I noticed, `detail` doesn't currently contain `omitempty`. Is that intended?

File: scripts/rules.go (hunk 1 above)
Comment (Emyrk) [reply scripts/rules.go:124]: Lol my bad

**Comment by Emyrk** [comment]
> httpapi.Response

not a bad idea to add it. Adding it now :+1:

File: coderd/coderd.go [hunk 2]
```diff
@@ -123,6 +123,7 @@ func New(options *Options) *API {
 		)
 		r.Get("/", func(w http.ResponseWriter, r *http.Request) {
 			httpapi.Write(w, http.StatusOK, httpapi.Response{
+				//nolint:gocritic
```
[the remark this answers is not in the document]
Comment (Emyrk) [reply coderd/coderd.go:126]: The only message I needed to make an exception for :smile:

**Pull request merged**